kolibri/content/test/test_content_app.py

```diff
@@ -0,0 +1,293 @@
+"""
+To run this test, type this in command line <kolibri manage test -- kolibri.content>
+"""
+import os
+import shutil
+import tempfile
+from django.test import TestCase
+from django.db import connections, IntegrityError
+from django.test.utils import override_settings
+from kolibri.content import models as content
+from kolibri.content import api
+from django.conf import settings
+
+
+@override_settings(
+    CONTENT_COPY_DIR=os.path.dirname(os.path.dirname(os.path.abspath(__file__)))+"/test_content_copy"
+)
+class ContentMetadataTestCase(TestCase):
+    """
+    Testcase for content API methods
+    """
+    fixtures = ['channel_test.json', 'content_test.json']
+    multi_db = True
+    the_channel_id = 'content_test'
+    connections.databases[the_channel_id] = {
+        'ENGINE': 'django.db.backends.sqlite3',
+        'NAME': ':memory:',
+    }
+
+    def setUp(self):
+        # Create a temporary directory
+        self.test_dir = tempfile.mkdtemp()
+        # Create files in the temporary directory
+        self.temp_f_1 = open(os.path.join(self.test_dir, 'test_1.pdf'), 'w')
```

Comment:
We found while setting a temp location for KOLIBRI_HOME for testing that it
might be nicer to have a consistent path, like cwd + .kolibri_home

On 15 February 2016 at 04:37, Richard Tibbles <notifications@github.com>
wrote:

> In kolibri/content/test/test_content_app.py
> <https://github.com/learningequality/kolibri/pull/25#discussion_r52857065>
> :
>
> > +    """
> > +    Testcase for content API methods
> > +    """
> > +    fixtures = ['channel_test.json', 'content_test.json']
> > +    multi_db = True
> > +    the_channel_id = 'content_test'
> > +    connections.databases[the_channel_id] = {
> > +        'ENGINE': 'django.db.backends.sqlite3',
> > +        'NAME': ':memory:',
> > +    }
> > +
> > +    def setUp(self):
> > +        # Create a temporary directory
> > +        self.test_dir = tempfile.mkdtemp()
> > +        # Create files in the temporary directory
> > +        self.temp_f_1 = open(os.path.join(self.test_dir, 'test_1.pdf'), 'w')
>
> [image: image]
> <https://cloud.githubusercontent.com/assets/1680573/13039330/5fa64818-d352-11e5-9a80-6cb8035dcffb.png>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/learningequality/kolibri/pull/25/files#r52857065>.
>